lib/internal/errors.js

```diff
@@ -185,12 +186,26 @@ E('ERR_V8BREAKITERATOR', 'Full ICU data not installed. ' +
 function invalidArgType(name, expected, actual) {
   const assert = lazyAssert();
   assert(name, 'name is required');
-  const type = name.includes('.') ? 'property' : 'argument';
-  var msg = `The "${name}" ${type} must be ${oneOf(expected, 'type')}`;
-  if (arguments.length >= 3) {
-    msg += `. Received type ${actual !== null ? typeof actual : 'null'}`;
+  assert.strictEqual(arguments.length, 3, 'arguments length mismatch');
+  const argType = name.includes('.') ? 'property' : 'argument';
+  // Only check the first entry
+  const expectedType = Array.isArray(expected) ? expected[0] : expected;
+  // A-Z = 65-90, a-z = 97-122
+  const type = expectedType.charCodeAt(0) > 96 ? 'of type' : 'instance of';
+  var received;
+  if (actual == null) {
+    received = String(actual);
+  } else if (typeof actual === 'object') {
+    if (actual.constructor != null) {
+      received = `instance of ${actual.constructor.name}`;
```

Comment:
Provided that ``actual`` is an arbitrary user-provided object which we are *validating*, after this patch, exceptions can be raised on any of:
- ``actual.constructor`` access.
- ``actual.constructor.name`` access.
- String concatenation with ``actual.constructor.name`` (if it's a ``Symbol`` or ``String(actual.constructor.name)`` throws)